aws/logs_monitoring/lambda_function.py

```diff
@@ -650,9 +650,21 @@ def add_metadata_to_lambda_log(event):
     function_name = lambda_log_arn.split(':')[6]
 
     event[DD_HOST] = lambda_log_arn
-    event[DD_CUSTOM_TAGS] = "{},functionname:{}".format(event[DD_CUSTOM_TAGS], function_name)
     event[DD_SERVICE] = function_name
 
+    tags = ['functionname:{}'.format(function_name)]
+
+
+    # Add any enhanced tags from metadata
+    if DD_ENHANCED_LAMBDA_METRICS:
```

Comment:
nit: thinking about it more I think the name `DD_ENHANCED_LAMBDA_METRICS` doesn't make as much sense as I originally thought, since the variable is really representing whether the enhanced metrics file is present. I think something like `IS_ENHANCED_METRICS_FILE_PRESENT` would be more descriptive but we don't necessarily need to change it in this PR

Reply:
It's a small enough change, I'll add it here.

Reply:
thanks